File: 231116/만나는 그 순간/the-moment-we-meet.py
N,M=map(int,input().split())
A=[0]*30
B=[0]*30

At,Ad=1,0
for _ in range(N):
    direction,time=input().split()
    time=int(time)
    for t in range(At,At+time):
        if direction=='R':
            Ad+=1
            A[t]=Ad
        else:
            Ad-=1
            A[t]=Ad
    At+=time

Bt,Bd=1,0
for _ in range(M):
    direction,time=input().split()
    time=int(time)
    for t in range(Bt,Bt+time):
        if direction=='R':
            Bd+=1
            B[t]=Bd
        else:
            Bd-=1
            B[t]=Bd
    Bt+=time

flag=0
for i in range(1,1001):
    if A[i]==B[i]:
        flag=1
        print(i)
        break
    
if flag==0:
    print(-1)

File: 251209/만나는 그 순간/the-moment-we-meet.js
const fs =require("fs");
const input = fs.readFileSync(0).toString().trim().split('\n');
const [n,m] = input[0].split(' ').map(Number);

const movesA = input.slice(1,1+n).map((move)=>move.split(' '));
const movesB = input.slice(1+n).map((move)=>move.split(' '));

const MAX_R = 2000000;

const arrA = Array(MAX_R+1).fill(0);
const arrB = Array(MAX_R+1).fill(0);

let timeA =0, curA =0;
for(let move of movesA){
    let [d, x] = move;
    x = Number(x);
    if(d ==='R'){
        while(x){
            curA++;
            timeA++;
            arrA[timeA] = curA;
            x--;
        }
    }else{
        while(x){
            curA--;
            timeA++;
            arrA[timeA] = curA;
            x--;
        }
    }
}

let timeB =0, curB =0;
for(let move of movesB){
    let [d, x] = move;
    x = Number(x);
    if(d ==='R'){
        while(x){
            curB++;
            timeB++;
            arrB[timeB] = curB;
            x--;
        }
    }else{
        while(x){
            curB--;
            timeB++;
            arrB[timeB] = curB;
            x--;
        }
    }
}

let flag = false;
let ans = 0;
for(let i=1; i<timeA+1; i++){
    if(arrA[i]===arrB[i]){
        console.log(arrA[i],arrB[i])
        flag=true;
        ans=i;
        break;
    }
}

console.log(flag?ans:-1)

